src/plugins/intel_cpu/tests/functional/shared_tests_instances/snippets/mlp_seq.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/mlp.hpp"
+
+#include "utils.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+
+namespace {
+
+std::vector<std::vector<InputShape>> inputShape_2D(bool with_dynamic = true) {
+    auto shapes = SNIPPETS_TESTS_STATIC_SHAPES(
+        {{1, 64}},
+        {{2, 64}},
+        {{4, 64}},
+        {{8, 64}});
+    if (with_dynamic) {
+        shapes.push_back({{PartialShape{-1, 64}, {{1, 64}, {8, 64}, {8, 64}, {6, 64}}}});
+        shapes.push_back({{PartialShape{-1, 64}, {{2, 64}, {2, 64}, {4, 64}, {3, 64}}}});
+    }
+    return shapes;
+}
+
+// Returns a vector of pairs where:
+//   - The first element is the number of hidden layers in the MLP
+//   - The second element is a pair: {expected number of subgraphs, expected number of nodes}
+std::vector<std::pair<size_t, std::pair<size_t, size_t>>> numHiddenLayersWithExpectations() {
+    return {
+        {1, {1, 1}},
+        {3, {2, 2}},
+        {5, {3, 3}},
+    };
+}
+
+std::vector<std::pair<size_t, std::pair<size_t, size_t>>> numHiddenLayersWithExpectationsBf16() {
+    return {
+        {1, {3, 9}},
+        {3, {3, 13}},
+        {5, {5, 18}},
+    };
+}
+
+std::vector<std::pair<size_t, std::pair<size_t, size_t>>> numHiddenLayersWithExpectationsQuantized() {
+    return {
+        {1, {1, 1}},
+        {3, {1, 1}},
+        {5, {1, 1}},
```

Comment:
🤔 This looks suspicious: actually we don't have enough registers to handle 5 Matmuls with postops in one Subgraph. Let's discuss it offline

Reply:
Actually, yes, after fixes 5 matmuls fit in 2 subgraphs only